examples/webgl/src/lib.rs

```diff
@@ -44,17 +45,18 @@ pub fn draw() {
     let program = link_program(&context, [vert_shader, frag_shader].iter()).unwrap();
     context.use_program(Some(&program));
 
-    let vertices = [-0.7, -0.7, 0.0, 0.7, -0.7, 0.0, 0.0, 0.7, 0.0];
-    let vert_array = js_sys::Float32Array::new(&wasm_bindgen::JsValue::from(vertices.len() as u32));
-    for (i, f) in vertices.iter().enumerate() {
-        vert_array.fill(*f, i as u32, (i + 1) as u32);
-    }
+    let vertices: [f32; 9] = [-0.7, -0.7, 0.0, 0.7, -0.7, 0.0, 0.0, 0.7, 0.0];
+    let memory_buffer = wasm_bindgen::memory().dyn_into::<WebAssembly::Memory>().unwrap().buffer();
+    let vert_array = js_sys::Float32Array::new(&memory_buffer).subarray(
+        &vertices as *const f32 as u32 / 4,
```

Comment:
Could this use `vertices.as_ptr()` instead of raw pointer casts?

Reply:
With the `new Uint8Array(buffer, byteOffset, length)` constructor it could be simplified to

```
let vert_array = js_sys::Uint8Array::new(&memory_buffer, vertices.as_ptr() as u32, vertices.len() * mem::size_of::<f32>());
```

That constructor doesn't seem to be exposed in js_sys currently though.

Reply:
Updated to use as_ptr() and tested on my side project.
Turns out the generated wasm is identical, even extracting verticles_location made no difference.

As a novice user, I was confused that I couldn't use the constructor @jsheard mentioned (or the float equivalent), but then I found the subarray work-around.